Improved TensorQTL.ipynb_phenotype error handling completed and singular matrix error in APOE interaction part

rl3328 opened the pull request:
I modified the read-in process of phenotype data and also changed the path of phenotype data in aws. Hao did the ROSMAP mQTL, haQTL in cluster. So, there's something to customize. I am testing the pipeline using Knight_mQTL, but since this QTL is analyzed all chromosomes all  at once, it might take a long time.

gaow commented:
1. remove `print` statement or make that into proper log messages
2. please include updates to pandas related code to use its latest version

rl3328 commented:
For error handling process for phenotype data, modifications are as follow:
1. For QTLs(gpQTL, sQTL) whose id are not consisted of emsemble_id(ENSG...), there are  occasions where one row is not uniquely identified by ['chr', 'ID'] in cis_window file. I made multiple rows with the same ['chr', 'ID'] into one row, with the union start-end position.
2. Some phenotype data might has variants that don't exist in cis_window list, remove them and made phenotype data a subset of cis_window list, based on ['chr', 'ID'].
3. The phenotype data and cis_window file might have different start-end positions. When merging them, there will be 'start_x', 'start_y' if colnames are the same. handling this error by dropping the phenotype start and end columns(genomic locus).
4. For QTLs whose start-end positions are the same, the columns are modified into pos column when using `read_phenotype_bed()` in tensorqtl package.  take this into consideration.

rl3328 commented:
For new setup customization,
1. `squeeze` usage is updated to catch up with new version of pandas
2. entrypoints and were removed

rl3328 commented:
pipeline in xqtl-protocol/code/ and pipeline in xqtl-protocol/pipeline/ are the same.

gaow commented on code/association_scan/TensorQTL/TensorQTL.ipynb:
```diff
@@ -565,18 +564,21 @@
     "    if \"#chr\" in molecular_pheno_files.columns:\n",
-    "        molecular_pheno_files = molecular_pheno_files.groupby(['#chr', '#dir']).size().reset_index(name='count').drop(\"count\",axis = 1).rename(columns = {\"#chr\":\"#id\"})\n",
+    "        molecular_pheno_files = molecular_pheno_files.groupby(['#chr','path']).size().reset_index(name='count').drop(\"count\",axis = 1).rename(columns = {\"#chr\":\"#id\"})\n",
+    "        print(molecular_pheno_files.head())\n",
```
Please remove this `print` statement

gaow commented on code/association_scan/TensorQTL/TensorQTL.ipynb:
```diff
@@ -565,18 +564,21 @@
     "    input_files = input_files.values.tolist()\n",
     "    input_chroms = [x[0] for x in input_files]\n",
-    "    input_files = [x[1:] for x in input_files]"
+    "    print(input_files)\n",
```
again, if you want to keep track of messages you should use formal logging (`logging` module in Python) not just `print`. Pleease clean it up

gaow commented on pipeline/TensorQTL.ipynb:
Is this file just a duplicate of what's in `code`? if so please remove it. Files in `pipeline` are symbolic links generated automatically running `release.sos`. It was created for convenience at the stage when we develop the protocol we may create different subfolders to organize the codes which keeps changing so we use links to ensure backwards compatiblity. Now as the protocol gets more stable I see less of a necessity using this folder. Maybe at some point we will just delete this folder altogether. 

So , at this point please don't add the same duplicated codes into this folder.

rl3328 replied:
Sounds great! But one of my concern is that, if we delete the pipeline folder altogether, when executeing `sos run xqtl-protocol/association_scan/TensorQTL/TensorQTL.ipynb`, it will return error: fail to locate xqtl-protocol/association_scan/TensorQTL/TensorQTL.ipynb.sos

rl3328 commented:
1. I removed the print() in pipeline, but keep those print() that are in expansion block(python) in sos, which will be shown as message or error in the stdout file.
2. I removed the duplicated file in pipeline.

gaow commented:
@rl3328 thanks -- for what you removed / fixed please mark my review comment as "resolved" that's easier for me to review your new patches

rl3328 commented:
package fixed, pipeline unchanged but can work fine now

rl3328 commented:
To resolve the negative 'pvalue' issue, I removed those rows whose 'pval_gi' is null(tstat won't convert into pvalue).

gaow commented:
@rl3328 thanks for keep fixing it up. Is it still WIP (in which case please put that back into the title of the PR) or it is now ready to be reviewed/accepted?

rl3328 commented:
Yes, both PR in xqtl-protocol and xqtl-analysis are ready for review!
